tests/foreman/api/test_discoveryrule.py

```diff
@@ -23,221 +23,101 @@
-        self.discovery_rule = entities.DiscoveryRule(
-            hostgroup=self.hostgroup, search_=gen_choice(searches)
-        )
-        self.discovery_rule._fields['hostname'].default = 'myhost-<%= rand(99999) %>'
-
-    @tier1
-    def test_positive_create_with_name(self):
-        """Create a new discovery rule.
-
-        Set query as (e.g CPU_Count = 1)
-
-        :id: b8ae7a80-b9a8-4924-808c-482a2b4102c4
-
-        :expectedresults: Rule should be created with given name and query
-
-        :CaseImportance: Critical
-        """
-        for name in valid_data_list():
-            with self.subTest(name):
-                self.discovery_rule.name = name
-                discovery_rule = self.discovery_rule.create()
-                self.assertEqual(self.discovery_rule.name, discovery_rule.name)
-                self.assertEqual(self.discovery_rule.search_, discovery_rule.search_)
-
-    @tier2
-    def test_positive_create_with_org_loc(self):
-        """Create discovery rule by associating org and location
-
-        :id: 121e0a30-8a24-47d7-974d-998886ed1ea7
-
-        :expectedresults: Rule was created and with given org & location.
-
-        :CaseLevel: Component
-        """
-        org = entities.Organization().create()
-        loc = entities.Location().create()
-        hostgroup = entities.HostGroup(organization=[org]).create()
-        discovery_rule = entities.DiscoveryRule(
-            hostgroup=hostgroup, search_='cpu_count = 1', organization=[org], location=[loc]
-        ).create()
-        self.assertEqual(org.name, discovery_rule.organization[0].read().name)
-        self.assertEqual(loc.name, discovery_rule.location[0].read().name)
-
-    @tier1
-    def test_positive_delete(self):
-        """Delete a discovery rule
-
-        :id: 9fdba953-dcc7-4532-9204-17a45b0d9e05
-
-        :expectedresults: Rule should be deleted successfully
-
-        :CaseImportance: Critical
-        """
-        for name in valid_data_list():
-            with self.subTest(name):
-                self.discovery_rule.name = name
-                discovery_rule = self.discovery_rule.create()
-                discovery_rule.delete()
-                with self.assertRaises(HTTPError):
-                    discovery_rule.read()
+        name = gen_choice(valid_data_list())
+
+        with self.subTest(name):
+            # Create discovery rule
+            search = gen_choice(searches)
+            hostname = 'myhost-<%= rand(99999) %>'
+            org = entities.Organization().create()
+            loc = entities.Location().create()
+            hostgroup = entities.HostGroup(organization=[org]).create()
+            discovery_rule = entities.DiscoveryRule(
+                name=name,
+                search_=search,
+                hostname=hostname,
+                organization=[org],
+                location=[loc],
+                hostgroup=hostgroup,
+            ).create()
+            self.assertEqual(name, discovery_rule.name)
```

Comment:
One suggestion from my end tries to use python assert rather than self. assert, As we already in phase moving that from PR  https://github.com/SatelliteQE/robottelo/pull/7686

Reply:
Okay it's fixed, thanks for pointing that out.